Vic2ToHoI4/Source/HOI4World/HoI4Country.cpp

```diff
@@ -276,6 +277,182 @@ void HoI4::Country::convertLeaders(const graphicsMapper& theGraphics, const Vic2
 }
 
 
+void HoI4::Country::convertMonarchIdea(const graphicsMapper& theGraphicsmapper,
+	 Names& names,
+	 Localisation& hoi4Localisations,
+	 const Vic2::Country& sourceCountry)
+{
+	if (!hasMonarchIdea())
+	{
+		return;
+	}
+
+	std::optional<std::string> firstName;
+	std::optional<std::string> surname = sourceCountry.getLastDynasty();
+
+	bool female = false;
+	if (female)
+	{
+		firstName = names.getFemaleName(primaryCulture);
+		auto femaleSurname = names.getFemaleSurname(primaryCulture);
+		if (!firstName)
+		{
+			firstName = names.getMaleName(primaryCulture);
+			if (!surname)
+			{
+				surname = names.getSurname(primaryCulture);
+			}
+			female = false;
+		}
+		else if (!surname)
+		{
+			if (femaleSurname)
+			{
+				surname = femaleSurname;
+			}
+			else
+			{
+				surname = names.getSurname(primaryCulture);
+			}
+		}
+	}
+	else
+	{
+		firstName = names.getMaleName(primaryCulture);
+		const auto newSurname = names.getSurname(primaryCulture);
+
+		if (!firstName)
+		{
+			firstName = names.getFemaleName(primaryCulture);
+			auto femaleSurname = names.getFemaleSurname(primaryCulture);
+			if (!newSurname && femaleSurname)
+			{
+				surname = femaleSurname;
+			}
+			female = true;
```

Comment:
You're assigning female true regardless of wheather firstName returned something?

Reply:
Yes. Note that this only happens when we already couldn't get a male name. If we also can't get a female name, we'll completely abort below.